tests/functional/test_application_restricted.py

```diff
@@ -0,0 +1,232 @@
+import datetime
+import jwt
+import uuid
+import time
+import requests
+from .config_files import config
+from pytest import mark
+from pytest_bdd import scenario, given, when, then, parsers
+
+
+# simple patient request
+def get_patient_request(headers: dict):
+    return requests.get(
+        f"{config.BASE_URL}/{config.PDS_BASE_PATH}/Patient?",
+        headers=headers,
+        params={
+            "family": "Smith",
+            "gender": "female",
+            "birthdate": "eq2010-10-22"
+        }
+    )
+
+
+@mark.skip(reason="unfinished code")
+@scenario(
+    "features/application_restricted.feature",
+    "PDS FHIR API accepts request with valid access token",
+)
+def test_valid():
+    pass
+
+
+@mark.skip(reason="unfinished code")
+@scenario(
+    "features/application_restricted.feature",
+    "PDS FHIR API rejects request with invalid access token",
+)
+def test_invalid():
+    pass
+
+
+@mark.skip(reason="unfinished code")
+@scenario(
+    "features/application_restricted.feature",
+    "PDS FHIR API rejects request with missing access token",
+)
+def test_missing():
+    pass
+
+
+@mark.skip(reason="unfinished code")
+@scenario(
+    "features/application_restricted.feature",
+    "PDS FHIR API rejects request with expired access token",
+)
+def test_expired():
+    pass
+
+
+@given("I am authenticating using unattended access", target_fixture="auth")
+def auth():
+    return {}
+
+
+@given("I have a valid access token")
+def set_valid_access_token(auth):
+    claims = {
+        "sub": config.APPLICATION_RESTRICTED_API_KEY,
+        "iss": config.APPLICATION_RESTRICTED_API_KEY,
+        "jti": str(uuid.uuid4()),
+        "aud": f"{config.BASE_URL}/oauth2/token",
+        "exp": int(time.time()) + 300,
+    }
+
+    headers = {
+        "kid": config.KEY_ID
+        }
+
+    encoded_jwt = jwt.encode(claims, config.SIGNING_KEY, algorithm="RS512", headers=headers)
+
+    response = requests.post(
+        f"{config.BASE_URL}/oauth2/token",
+        data={
+            "grant_type": "client_credentials",
+            "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+            "client_assertion": encoded_jwt,
+        },
+    )
+
+    response_json = response.json()
+
+    assert {"access_token", "expires_in", "token_type"} == response_json.keys()
+    assert response_json["access_token"] is not None
+    assert response_json["token_type"] == "Bearer"
+    try:
+        response_expires_in = int(response_json["expires_in"])
+        assert response_expires_in >= 0
+    except ValueError:
```

Comment:
I think you can probably leave this out tbh, and just assert that it's a valid timestamp (not empty or None) and it's sometime in the future